iDiff/cmd/root.go

```diff
@@ -27,35 +27,44 @@ var RootCmd = &cobra.Command{
 			glog.Error(err.Error())
 			os.Exit(1)
 		}
+
 		utils.SetDockerEngine(eng)
-		image1, err := utils.ImagePrepper{args[1]}.GetImage()
+
+		img1Arg := args[len(args)-2]
+		img2Arg := args[len(args)-1]
+		diffArgs := args[:len(args)-2]
+
+		image1, err := utils.ImagePrepper{img1Arg}.GetImage()
 		if err != nil {
 			glog.Error(err.Error())
 			os.Exit(1)
 		}
-		image2, err := utils.ImagePrepper{args[2]}.GetImage()
+		image2, err := utils.ImagePrepper{img2Arg}.GetImage()
 		if err != nil {
 			glog.Error(err.Error())
 			os.Exit(1)
 		}
-		differ, err := differs.GetDiffer(args[0])
+		diffTypes, err := differs.GetDiffers(diffArgs)
 		if err != nil {
 			glog.Error(err.Error())
 			os.Exit(1)
 		}
 
-		diff := differs.DiffRequest{image1, image2, differ}
-		if diff, err := diff.GetDiff(); err == nil {
-			if json {
-				err = diff.OutputJSON()
-				if err != nil {
-					glog.Error(err)
-				}
-			} else {
-				err = diff.OutputText()
-				if err != nil {
-					glog.Error(err)
+		req := differs.DiffRequest{image1, image2, diffTypes}
+		if diffs, err := req.GetDiff(); err == nil {
+			for _, diff := range diffs {
+				if json {
+					err = diff.OutputJSON()
+					if err != nil {
+						glog.Error(err)
+					}
+				} else {
+					err = diff.OutputText()
```

Comment:
Currently the outputs come labeled with their DiffResult type when output as text...not sure if that was the intention or not but if you were to run pip and apt since they both return values of the same type, you would not be able to distinguish the two outputs.